pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -2006,13 +2007,29 @@ func buildTracingConfig(config *meshconfig.ProxyConfig, proxyType model.NodeType
 		},
 	}
 
+	if config.Tracing != nil && proxyType == model.SidecarProxy {
+		buildSidecarTracingConfig(config, proxyType, tracingCfg)
+	}
+
+	return tracingCfg
+}
+
+func buildSidecarTracingConfig(config *meshconfig.ProxyConfig, proxyType model.NodeType, tracingCfg *http_conn.HttpConnectionManager_Tracing) {
```

Comment:
once https://github.com/istio/istio/pull/23082 merges this will be wrong, so just need to be sure we update it whichever merges first

Reply:
Actually looks like your other PR already made #23082 wrong (it should use the proxyconfig from metadata, not from Pilot) so i'll update mine

Reply:
I'd prefer whatever you feel is a best approach code-wise. I had adapted the unit test from an already existing test. Let me know.

Reply:
I just mean that we need to make sure `config` comes from the node metadata rather than meshConfig.defaultConfig. Since a proxy can override it (once that PR merges)

Reply:
@howardjohn have you done the same for all other places which use ProxyConfig?

My suggestion would be to start using `ProxyConfig` from node metadata in one PR (if possible), easier to review and reason.

Reply:
When I looked earlier today there were no usages prior to this morning, when the other one for extra tags was merged